src/security/library.cpp

```diff
@@ -0,0 +1,585 @@
+#include "library.h"
+
+#include <string>
+
+extern "C" {
+#include <ngx_core.h>
+#include <ngx_cycle.h>
+#include <ngx_log.h>
+#include <ngx_string.h>
+}
+
+#include <ddwaf.h>
+#include <rapidjson/schema.h>
+
+#include <fstream>
+#include <numeric>
+#include <optional>
+#include <stdexcept>
+#include <string_view>
+#include <utility>
+
+#include "blocking.h"
+#include "context.h"
+#include "ddwaf_obj.h"
+#include "util.h"
+
+extern "C" {
+#define INCBIN_SILENCE_BITCODE_WARNING
+#include "incbin.h"
+}
+
+extern "C" {
+INCBIN(char, RecommendedJson, "security/recommended.json");
+}
+
+using namespace std::literals;
+namespace dnsec = datadog::nginx::security;
+
+namespace {
+
+static constexpr auto kDefaultKeyRegex =
```

Comment:
There is a default defined here and one in `FinalizedConfigSettings`, you could get rid of one probably.

Reply:
Indeed. I also see that I considered emptiness and non-presence of the environment variable the same, which shouldn't be done.